pkg_resources/__init__.py

```diff
@@ -2608,11 +2608,20 @@ def hashcmp(self):
             self.parsed_version,
             self.precedence,
             self.key,
-            _remove_md5_fragment(self.location),
+            self._location_without_md5,
             self.py_version or '',
             self.platform or '',
         )
 
+    def get_location(self):
+        return self._location
+
+    def set_location(self, l):
+        self._location = l
+        self._location_without_md5 = _remove_md5_fragment(l)
+
+    location = property(get_location, set_location)
```

Comment:
I wonder if this is to provide some sort of compatibility, or the following is actually OK:
```suggestion
    @property
    def location(self):
        return self._location

    @location.setter
    def location(self, location):
        self._location = location
        self._location_without_md5 = _remove_md5_fragment(location)
```

Reply:
Let's test it as well. I do not have a preference.